src/idl_parser.cpp

```diff
@@ -1758,6 +1771,12 @@ CheckedError Parser::ParseSingleValue(const std::string *name, Value &e,
         TRY_ECHECK(kTokenStringOrIdent, IsBool(in_type), BASE_TYPE_BOOL);
       }
     }
+    // Check for optional scalars.
+    if (!match && IsScalar(in_type) && attribute_ == "null") {
+      e.constant = "null";
+      NEXT();
+      match = true;
```

Comment:
Did you check that all callers of `ParseSingleValue` can deal with `"null"`, for example JSON parsing?

Reply:
I agree, it is needed to check the parsing and printing of JSON.
1.  `{ "opt_field" : 123 }`
2.  `{ "opt_field" : 4.56 }`
3.  `{ "opt_field" : "null" }`
4.  `{}` - implicit null

Reply:
Added a test to parse Json to Flexbuffers